src/v/resource_mgmt/cpu_profiler.cc

```diff
@@ -146,4 +149,33 @@ void cpu_profiler::on_sample_period_change() {
     ss::engine().set_cpu_profiler_period(_sample_period());
 }
 
+ss::future<std::vector<cpu_profiler::shard_samples>>
+cpu_profiler::override_and_get_results(
+  std::chrono::milliseconds timeout, std::optional<ss::shard_id> shard_id) {
+    if (_gate.is_closed()) {
+        co_return std::vector<shard_samples>{};
+    }
+    auto holder = _gate.hold();
```

Comment:
If you set a 15min profile time, holding this gate will prevent the server from being shut down for 15 min, right?

Maybe it's better to drop the gate and grab it against after the sleep (which may occasionally fail if it races with shutdown, which is fine).

Reply:
Good point, it is possible that it'll cause a segfault on shutdown though if the gate isn't held during the wait. I.e, the object is destroyed as a result of shutting down then the task tries to reacquire the gate.

Reply:
I could make the wait abort-able then have the shutdown logic abort any on-going waits. This would require keeping a list of abortables in the class, but I don't believe that's the end of the world.

Reply:
Good point about gate re-acquisition. Too bad `gate` doesn't support that scenario: safely checking whether the underlying gate object has gone away (e.g., via reference counting).

Your suggestion sounds good to me.